### PR TITLE
HUE-9728 [sqlAlchemy] Add views to get_tables API

### PR DESCRIPTION
**Pull request by agl29**
Tested manually
With adding `get_view_names()` ss -> https://drive.google.com/file/d/1FNAa6Hn5Fw2MEZstoOAQveJjOWwVOu2O/view?usp=sharing
without adding  ss->  https://drive.google.com/file/d/1MDFUoUkk5i6Jl1IxX8A92KtzH645Oona/view?usp=sharing

### CODE REVIEW

**Review by romainr — COMMENTED** [review]
Add a mock test?

https://github.com/cloudera/hue/blob/master/desktop/libs/notebook/src/notebook/connectors/sql_alchemy_tests.py#L260

**Comment by romainr** [comment]
Nice!

So now for example on demo.gethue we should start seeing the MySql views:

`CREATE view top_salaries as select * from emploees.salaries order by salary desc limit 5`

However, we should probably set the type as 'View', so that we show the little eye icon? (like in Hive)

https://github.com/cloudera/hue/blob/master/desktop/libs/notebook/src/notebook/connectors/sql_alchemy.py#L441

Let's get this PR in then switch to use the proper type to show them as views in the left assist.

**Review by romainr — APPROVED** [review]

File: desktop/libs/notebook/src/notebook/connectors/sql_alchemy.py [hunk 1]
```diff
@@ -438,9 +438,12 @@ def autocomplete(self, snippet, database=None, table=None, column=None, nested=N
...
     elif table is None:
       tables_meta = []
-      for t in assist.get_tables(database):
+      for t in assist.get_table_names(database):
         t = self._fix_bigquery_db_prefixes(t)
         tables_meta.append({'name': t, 'type': 'Table', 'comment': ''})
+      for t in assist.get_view_names(database):
+        t = self._fix_bigquery_db_prefixes(t)
+        tables_meta.append({'name': t, 'type': 'View', 'comment': ''})
       response['tables_meta'] = tables_meta
```
Comment (agl29) [inline desktop/libs/notebook/src/notebook/connectors/sql_alchemy.py:447]: This is working @romainr please review it.

File: desktop/libs/notebook/src/notebook/connectors/sql_alchemy_tests.py [hunk 2]
```diff
@@ -273,6 +273,22 @@ def test_get_sample_data(self):
...
+
+    with patch('notebook.connectors.sql_alchemy.create_engine') as create_engine:
+      with patch('notebook.connectors.sql_alchemy.inspect') as inspect:
+        with patch('notebook.connectors.sql_alchemy.Assist.get_table_names') as get_table_names:
+          with patch('notebook.connectors.sql_alchemy.Assist.get_view_names') as get_view_names:
+            get_table_names.return_value = ['table1']
+            get_view_names.return_value = ['view1']
+
+            response = SqlAlchemyApi(self.user, self.interpreter).autocomplete(snippet, database='database1')
+            assert_equal(response['tables_meta'][0]['name'], 'table1')
```
Comment (romainr) [inline desktop/libs/notebook/src/notebook/connectors/sql_alchemy_tests.py:288]: Add check of the 'type' value?

**Review by romainr — APPROVED** [review]

File: desktop/libs/notebook/src/notebook/connectors/sql_alchemy_tests.py (hunk 2 above)
Comment (agl29) [reply desktop/libs/notebook/src/notebook/connectors/sql_alchemy_tests.py:288]: Done

**Pull request merged**